packages/scoutgame/src/users/mergeUserAccount.ts

```diff
@@ -0,0 +1,215 @@
+import type { Prisma } from '@charmverse/core/prisma-client';
+import { prisma } from '@charmverse/core/prisma-client';
+
+export type ProfileToKeep = 'current' | 'new';
+
+export const mergeUserAccount = async ({
+  userId,
+  farcasterId,
+  telegramId,
+  selectedProfile
+}: {
+  userId: string;
+  farcasterId?: number;
+  telegramId?: number;
+  selectedProfile: ProfileToKeep;
+}) => {
+  if (!farcasterId && !telegramId) {
+    throw new Error('No account identities to merge');
+  }
+
+  const [primaryUser, secondaryUser] = await Promise.all([
+    prisma.scout.findUniqueOrThrow({
+      where: {
+        id: userId
+      },
+      select: {
+        builderStatus: true
+      }
+    }),
+    prisma.scout.findFirstOrThrow({
+      where: farcasterId ? { farcasterId } : { telegramId },
+      select: {
+        builderStatus: true,
+        id: true
+      }
+    })
+  ]);
+
+  if (primaryUser.builderStatus !== null && secondaryUser.builderStatus !== null) {
+    throw new Error('Can not merge two builder accounts');
+  }
```

Comment:
you should check if the secondaryUser.deletedAt is set, and maybe just throw an error. this avoids someone calling it twice, but the 2nd time the secondary user would not have the correct attributes anymore

Reply:
If we return success it will fire off another function to reconcile the stats. Best would be to throw an error as this is not going to be possible from the frontend